esmf_regrid/_esmf_sdo.py

```diff
@@ -138,21 +149,37 @@ def __init__(
         self.circular = circular
         self.areas = areas
         super().__init__(
-            shape=(len(lats), len(lons)),
+            shape=shape,
             index_offset=1,
             field_kwargs={"staggerloc": ESMF.StaggerLoc.CENTER},
         )
 
     def _as_esmf_info(self):
         shape = np.array(self._shape)
 
-        if self.circular:
-            adjustedlonbounds = self.lonbounds[:-1]
-        else:
-            adjustedlonbounds = self.lonbounds
-
-        centerlons, centerlats = np.meshgrid(self.lons, self.lats)
-        cornerlons, cornerlats = np.meshgrid(adjustedlonbounds, self.latbounds)
+        londims = len(self.lons.shape)
+        assert len(self.lons.shape) == londims
```

Comment:
Why are we repeating all the shape checking? Is there a case where it is not caught in `__init__`? i.e. are `self.lons` and `self.lats` assigned to / changed elsewhere?

Reply:
If it's necessary, should probably be pulled out into e.g. `self._check_dim_shape()`.

Reply:
I think you're right that this is redundant. this information shouldn't be changing from initialisation to here.